src/data/api.js

```
@@ -161,6 +150,24 @@ export async function getCourseBlocks(courseId) {
   return normalizeBlocks(courseId, data.blocks);
 }
 
+export const getOutlineTabData = async (courseId, tab, version) => {
+  const url = `${getConfig().LMS_BASE_URL}/api/course_home/${version}/${tab}/${courseId}`;
+  try {
+    const { data } = await getAuthenticatedHttpClient().get(url);
+    const courseBlocks = normalizeBlocks(courseId, data.course_blocks.blocks);
+    const courseTools = camelCaseObject(data.course_tools);
+    return { courseTools, courseBlocks };
+  } catch (error) {
+    const { httpErrorStatus } = error && error.customAttributes;
+    if (httpErrorStatus === 404) {
+      return window.location.replace(`${getConfig().LMS_BASE_URL}/courses/${courseId}/dates`);
+    }
+    // async functions expect return values. to satisfy that requirement
+    // we return true here which in turn continues with the normal flow of displaying
+    // the "unexpected error try again" screen to the user.
+    return true;
```

The return values are all over the place here. You return a dictionary object in the happy path. An undefined object in the 404 case (window.location.replace doesn't return anything). And a bool in the error case. Can you bubble up the error here instead of returning anything? Or at least return an empty object to keep the types constant.

---

I'm realizing this redirect only exists for this case: https://github.com/edx/frontend-app-learning/pull/75 (redirecting back to the platform dates tab if waffle flags aren't enabled for the MFE dates tab)

Is this redirect even necessary for the outline tab? (ie. should I nix the try/catch logic)